[chore] Split up `cmd/goreleaser/internal/configure.go`

douglascamata opened the pull request:
Doing a little house keeping that I promised a very long time ago. This breaks up the huge `cmd/goreleaser/internal/configure.go` into many files, so that it's easy to find things and scroll around, particularly when it comes to looking at how we build the goreleaser configuration file for each one of the distros.

I'm very open to feedback on the names of the new files. Naming things is difficult. 

Small note: I generate the goreleaser files of all distros during my work to ensure that these changes still produces the exact same files as before.

mowies commented on cmd/goreleaser/internal/builder.go:
```diff
@@ -0,0 +1,512 @@
+	return b
+}
+
+func (b *distributionBuilder) dockerSigns() []config.Sign {
```
```suggestion
func (b *distributionBuilder) newDockerSigns() []config.Sign {
```

mowies commented on cmd/goreleaser/internal/builder.go:
```diff
@@ -0,0 +1,512 @@
+	return b
+}
+
+func (b *distributionBuilder) binaryRelease(header string) config.Release {
```
```suggestion
func (b *distributionBuilder) newBinaryRelease(header string) config.Release {
```

mowies commented on cmd/goreleaser/internal/distro_opamp.go:
```diff
@@ -0,0 +1,59 @@
+		withDefaultBinaryRelease(opampReleaseHeader).
+		withDefaultMSIConfig().
+		withDefaultNfpms().
+		// This is required because of same non-obvious path/workdir handling in
```
```suggestion
		// This is required because of some non-obvious path/workdir handling in
```

douglascamata commented:
Thanks for the review, @mowies. I addressed your feedback on commit [5336f2e](https://github.com/open-telemetry/opentelemetry-collector-releases/pull/1235/commits/5336f2e632192aa7a9281de413ce0c6c9dff8999).

songy23 reviewed (approved):
Looks great, thanks @douglascamata
